products/messaging/frontend/campaignsLogic.ts

```diff
@@ -0,0 +1,65 @@
+import { actions, kea, path, reducers, selectors } from 'kea'
+import { urlToAction } from 'kea-router'
+import { Scene } from 'scenes/sceneTypes'
+import { urls } from 'scenes/urls'
+
+import { Breadcrumb } from '~/types'
+
+import type { campaignsLogicType } from './campaignsLogicType'
+
+export const campaignsLogic = kea<campaignsLogicType>([
+    path(['products', 'messaging', 'frontend', 'campaignsLogic']),
+    actions({
+        editCampaign: (id: string | null) => ({ id }),
+    }),
+    reducers({
+        campaignId: [null as string | null, { editCampaign: (_, { id }) => id }],
+    }),
+    selectors({
+        breadcrumbs: [
+            (s) => [s.campaignId],
+            (campaignId): Breadcrumb[] => {
+                return [
+                    {
+                        key: Scene.MessagingCampaigns,
+                        name: 'Messaging',
+                        path: urls.messagingCampaigns(),
+                    },
+                    {
+                        key: 'campaigns',
+                        name: 'Campaigns',
+                        path: urls.messagingCampaigns(),
+                    },
+                    ...(campaignId === 'new'
+                        ? [
+                              {
+                                  key: 'new-campaign',
+                                  name: 'New campaign',
+                                  path: urls.messagingCampaignNew(),
+                              },
+                          ]
+                        : campaignId
+                        ? [
+                              {
+                                  key: 'edit-campaign',
+                                  name: 'Manage campaign',
+                                  path: urls.messagingCampaign(campaignId),
+                              },
+                          ]
+                        : []),
+                ]
+            },
+        ],
+    }),
+    urlToAction(({ actions }) => ({
+        '/messaging/campaigns/new': () => {
```

Comment:
Small one but from what I understand its better to use the `urls` here if you can as that way if you later change the url these kinds of things keep working. We don't do it everywhere sadly but its a handy thing to do up front and avoid future bugs